model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/JpaMapKeycloakTransaction.java

```diff
@@ -174,10 +170,10 @@ public Stream<E> read(QueryParameters<M> queryParameters) {
         TypedQuery<RE> emQuery = paginateQuery(em.createQuery(query), queryParameters.getOffset(), queryParameters.getLimit());
 
         Map<QueryCacheKey, List<RE>> cache = getQueryCache();
-        QueryCacheKey queryCacheKey = new QueryCacheKey(emQuery, modelType);
+        QueryCacheKey queryCacheKey = null;
+        queryCacheKey = new QueryCacheKey(queryParameters, modelType);
```

Comment:
Thank you @ahus1, I like the way it's done. Great idea.

I have just one nitpick: there is `null` assigned into `queryCacheKey` on line 173 and on 174 there is actual `QueryCacheKey`. Would it make sense:
```suggestion
        QueryCacheKey queryCacheKey = new QueryCacheKey(queryParameters, modelType);
```
?

Reply:
Will do. As Hynek suggested, I'm now preparing a new PR against main, see #16516 (it is currently in draft)